packages/dht/src/dht/IPeerManager.ts

```diff
@@ -0,0 +1,13 @@
+import { PeerIDKey } from '../helpers/PeerID'
+import { PeerDescriptor } from '../proto/packages/dht/protos/DhtRpc'
+import { DhtPeer } from './DhtPeer'
+
+export interface IPeerManager {
+    getClosestPeersTo(kademliaId: Uint8Array, limit?: number, excludeSet?: Set<PeerIDKey>): DhtPeer[]
+    getNumberOfPeers(excludeSet?: Set<PeerIDKey>): number
+    handlePeerActive(peer: DhtPeer): void
+    handlePeerUnresponsive(peer: DhtPeer): void
+    handleNewPeers(peerDescriptors: PeerDescriptor[], setActive?: boolean): void
+    getDistance(kademliaId1: Uint8Array, kademliaId2: Uint8Array): number
+    getKBucketSize(): number
```

Comment:
The handle methods could maybe renamed as it is but unclear what the methods do. Also method names don't need to contain the word peer as by default all methods of `PeerManager` operate on peers.

Some ideas, not sure if correct:

```suggestion
    getClosestTo(kademliaId: Uint8Array, limit?: number, excludeSet?: Set<PeerIDKey>): DhtPeer[]
    getCount(excludeSet?: Set<PeerIDKey>): number
    setActive(peer: DhtPeer): void
    onUnresponsive(peer: DhtPeer): void
    add(peerDescriptors: PeerDescriptor[], setActive?: boolean): void
    getDistance(kademliaId1: Uint8Array, kademliaId2: Uint8Array): number
    getKBucketSize(): number
```

Reply:
using "handle" is intetentional here, we do not want the caller to know what callee does when the event occurres. We just want that the callee to do something appropriate with it. This pattern is also used in some other places of the codebase, "handle" is kind of shorthand for handleXXEvent()

Reply:
We currently seem to use both `handleX` and `onX` naming conventions for event observers. Currently most of the classes in `dht` use the onX naming convention (`DhtNode`, `DiscoverySession`, `RecursiveFindSession`, `ConnectionManager`, `ManagedConnection`, `NodeWebRtcConnection`, `WebRtcConnector`) and in some classes the `handleX` convention is also used in one or two methods.

I'd prefer that we use the `onX` convention here and maybe migrate other usages to that convention in a separate PR later. (Or if we want to choose the other naming convention we can migrate all other use cases to that. But good to have just one naming convention.)

The `peer` part in the names is anyway redundant.

Reply:
The logic behind using the name prefixes "on" and "handle" are as follows: methods starting with "on" are event listeners that get invoked by eventemitter3, and they should not be called directly. Methods starting with "handle" are event handlers that can be called directly.

Reply:
Why a method should care who calls it? If it is a public method of `PeerManager`, any external component can call it. The `onFooChanged` pattern would clearly communicate that the external component should call this method whenever it changes the value of foo.